Zend/zend_partial.c

```diff
@@ -0,0 +1,932 @@
+    } else {
+		zend_get_gc_buffer *buffer = zend_get_gc_buffer_create();
+
+		if (Z_TYPE(partial->This) == IS_OBJECT) {
+			zend_get_gc_buffer_add_zval(buffer, &partial->This);
+		}
+
+		for (uint32_t arg = 0; arg < partial->argc; arg++) {
+			zend_get_gc_buffer_add_zval(buffer, &partial->argv[arg]);
+		}
+
+		zend_get_gc_buffer_use(buffer, table, n);
+    }
+
+	return NULL;
+}
+
+static zend_function *zend_partial_get_method(zend_object **object, zend_string *method, const zval *key) /* {{{ */
+{
+	if (zend_string_equals_literal_ci(method, ZEND_INVOKE_FUNC_NAME)) {
+		zend_partial *partial = (zend_partial*) *object;
+
+		return &partial->trampoline;
+	}
+
+	return zend_std_get_method(object, method, key);
+}
+/* }}} */
+
+static int zend_partial_get_closure(zend_object *obj, zend_class_entry **ce_ptr, zend_function **fptr_ptr, zend_object **obj_ptr, bool check_only)
+{
+	zend_partial *partial = (zend_partial*) obj;
+
+	*fptr_ptr = &partial->trampoline;
+	*obj_ptr  = (zend_object*) &partial->std;
+
+	return SUCCESS;
+}
+
+zend_function *zend_partial_get_trampoline(zend_object *obj) {
+	zend_partial *partial = (zend_partial*) obj;
+
+	return &partial->trampoline;
+}
+
+static void zend_partial_free(zend_object *object) {
+	zend_partial *partial = (zend_partial*) object;
+
+	zval *arg = partial->argv,
+		 *end = arg + partial->argc;
+
+	while (arg < end) {
+		if (Z_OPT_REFCOUNTED_P(arg)) {
+			zval_ptr_dtor(arg);
+		}
+		arg++;
+	}
+
+	efree(partial->argv);
+
+	if (partial->named) {
+		zend_array_release(partial->named);
+	}
+
+	zend_arg_info *info = partial->prototype.common.arg_info;
+
+	if (ZEND_PARTIAL_FUNC_FLAG(&partial->func, ZEND_ACC_HAS_RETURN_TYPE)) {
+		info--;
+	}
+
+	efree(info);
+
+	if (partial->func.type == ZEND_USER_FUNCTION) {
+		if (partial->prototype.op_array.filename) {
+			zend_string_release(partial->prototype.op_array.filename);
+		}
+		destroy_zend_function(&partial->func);
+	}
+
+	if (Z_TYPE(partial->This) == IS_OBJECT) {
+		zval_ptr_dtor(&partial->This);
+	}
+
+	zend_object_std_dtor(object);
+}
+
+void zend_partial_startup(void) {
+	memcpy(&zend_partial_handlers,
+			&std_object_handlers, sizeof(zend_object_handlers));
+
+	zend_partial_handlers.free_obj = zend_partial_free;
+	zend_partial_handlers.get_debug_info = zend_partial_debug;
+	zend_partial_handlers.get_gc = zend_partial_get_gc;
+	zend_partial_handlers.get_closure = zend_partial_get_closure;
+	zend_partial_handlers.get_method = zend_partial_get_method;
+
+	memset(&zend_call_magic_arginfo, 0, sizeof(zend_arg_info) * 1);
+
+	zend_call_magic_arginfo[0].name = ZSTR_KNOWN(ZEND_STR_ARGS);
+	zend_call_magic_arginfo[0].type = (zend_type) ZEND_TYPE_INIT_NONE(_ZEND_IS_VARIADIC_BIT);
+}
+
+static zend_always_inline zend_string* zend_partial_function_name(zend_function *function) {
+	if (function->type == ZEND_INTERNAL_FUNCTION) {
+		if (function->internal_function.handler == zend_pass_function.handler) {
+			return zend_strpprintf(0, "__construct");
```

Comment:
zend_string_init ^^

Reply:
Fixed in 9400a6930e5edbaba7052c2e9a0203816637de96